File: python/sort/bubbleSort.py
def bubbleSort(arr: list[int]) -> list[int]:
    result = arr.copy()

    while True:
        swapped = False

        for i in range(len(result) - 1):
            if result[i] > result[i + 1]:
                result[i], result[i + 1] = result[i + 1], result[i]
                swapped = True

        if not swapped:
            break

    return result


print(bubbleSort([10, 2, 5, 3, 1]))

File: js/sort/bubbleSort.js
// Big O(n^2)

function bubbleSort(arr) {
  let isContinue = false;

  do {
    isContinue = false;

    for (let i = 1; i < arr.length; i++) {
      const left = i - 1;
      const right = i;

      if (arr[left] > arr[right]) {
        [arr[left], arr[right]] = [arr[right], arr[left]];
        isContinue = true;
      }
    }
  } while (isContinue);

  return arr;
}

console.log(bubbleSort([10, 2, 5, 3, 1]));
